realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -23,6 +23,79 @@
  * element in the {@code OrderedRealmCollection} has an index. Each element can thus be accessed by its
  * index, with the first index being zero. Normally, {@code OrderedRealmCollection}s allow duplicate
  * elements, as compared to Sets, where elements have to be unique.
+ *
+ * <p>
+ *
+ * There are three types of {@link OrderedRealmCollection}. {@link RealmResults} and {@link RealmList} are live
+ * collections. They are up-to-date all the time and they will never contain a invalid {@link RealmObject}.
+ * {@link OrderedRealmCollectionSnapshot} is different. An {@link OrderedRealmCollectionSnapshot} can be created from
+ * another {@link OrderedRealmCollection}. Its size and elements order stay the same as the original collection's when
+ * it was created. {@link OrderedRealmCollection} may contain invalid {@link RealmObject} if the object gets deleted.
```

Comment:
OrderedRealmCollection -> OrderedRealmCollectionSnapshot 

{@link RealmObject} if the object gets deleted. -> `{@link RealmObject}'s if the objects gets deleted.`?